src/Pages/Dashboard/Dashboard.js

```diff
@@ -1,48 +1,31 @@
-import React, { useState, useContext } from "react";
-import "./dashboard.css";
-import Menu from "./SideMenu";
-import Survey from "./Survey/SurveyMenu";
-import Home from "./Home/Home";
-import Footprint from "./Footprint/Footprint";
-import Pledges from "./Pledges/Pledges";
-import News from "./News/News";
-// import Suggestions from "./Pledges/Suggestions";
-import Settings from "./Settings/Settings";
+import Stats from "./Stats/Stats";
+import React, { useContext, useState } from "react";
 import { UserContext } from "../../Main/UserContext";
+import FootprintSlideshow from "./FootprintSlideshow/FootprintSlideshow";
+import Pledges from "./Pledges/Pledges";
+import PledgesPage from "./PledgesPage/PledgesPage";
 
 export default () => {
-  const [menuPlacement, setMenuPlacement] = useContext(
-      UserContext
-    ).menuPlacement,
-    [wrapperClass, setWrapperClass] = useContext(UserContext).wrapperClass,
-    menuItems = [
-      { name: "Home", comp: <Home /> },
-      { name: "Your footprint", comp: <Footprint /> },
-      { name: "Pledges", comp: <Pledges /> },
-      { name: "News", comp: <News /> },
-      { name: "Survey", comp: <Survey /> },
-      {
-        name: "Options",
-        comp: (
-          <Settings setMenuPlacement={[setMenuPlacement, setWrapperClass]} />
-        ),
-      },
-    ],
-    [displayItem, setDisplayItem] = useState(menuItems[0].comp),
-    [user, setUsername] = useContext(UserContext).user;
-
-  // console.log(user)
+  const [pledgesRender, setpledgesRender] = useState(false);
+  const fb = useContext(UserContext).firebase;
```

Comment:
so we can use the Firebase API where we need it without importing the firebase.js. Since the UserProvider UserContext wraps the whole App in index.js. Is this a b ad approach, should we rather just import it where needed?

Reply:
How would you do it?

Reply:
Generally it's a bit weird to put everything in context (and in the same one), especially a library - if it works for now you can keep it though, but otherwise consider moving it out :)